src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -960,4 +1054,52 @@ public void visitToken(DetailAST ast) {
             throw new IndexOutOfBoundsException("test");
         }
     }
+
+    private static class DummyCheck extends AbstractFileSetCheck {
+
+        private final List<String> methodCalls = new ArrayList<>();
+
+        private boolean initCalled;
+
+        @Override
+        public void init() {
+            super.init();
+            initCalled = true;
+        }
+
+        @Override
+        public void beginProcessing(String charset) {
+            methodCalls.add("beginProcessing");
+            super.beginProcessing(charset);
+        }
+
+        @Override
+        public void finishProcessing() {
+            methodCalls.add("finishProcessing");
+            super.finishProcessing();
+        }
+
+        @Override
+        protected void processFiltered(File file, List<String> lines) throws CheckstyleException {
+            methodCalls.add("processFiltered");
+        }
+
+        @Override
+        public void destroy() {
+            methodCalls.add("destroy");
+            super.destroy();
+        }
+
+        public List<String> getMethodCalls() {
+            return Collections.unmodifiableList(methodCalls);
+        }
+
+        public boolean isInitCalled() {
+            return initCalled;
+        }
+
+        public MessageDispatcher messageDispatcherIsEqualsTo() {
```

Comment:
This is a weird method name for a getter. `getMessageDispatcher` is `final` and `protected` which is why we can't override it.
How about we name it `getInternalMessageDispatcher`?